qemu/tests/qemu_guest_agent.py

```diff
@@ -102,49 +101,45 @@ def _cleanup_open_session(self):
         except Exception:
             pass
 
-    @error.context_aware
-    def gagent_install(self, params, vm, *args):
+    def gagent_install(self, params, test, vm, *args):
         if args and isinstance(args, tuple):
             gagent_install_cmd = args[0]
         else:
-            raise error.TestError("Missing config 'gagent_install_cmd'")
+            test.error("Missing config 'gagent_install_cmd'")
 
         if not gagent_install_cmd:
             return
 
-        error.context("Try to install 'qemu-guest-agent' package.",
-                      logging.info)
+        logging.info("Try to install 'qemu-guest-agent' package.")
         session = self._get_session(params, vm)
         s, o = session.cmd_status_output(gagent_install_cmd)
         if s:
-            raise error.TestFail("Could not install qemu-guest-agent package"
-                                 " in VM '%s', detail: '%s'" % (vm.name, o))
+            test.fail("Could not install qemu-guest-agent package in VM '%s'",
+                      "detail: '%s'" % vm.name)
 
-    @error.context_aware
-    def gagent_start(self, params, vm, *args):
+    def gagent_start(self, params, test, vm, *args):
         if args and isinstance(args, tuple):
             gagent_start_cmd = args[0]
         else:
-            raise error.TestError("Missing config 'gagent_start_cmd'")
+            test.error("Missing config 'gagent_start_cmd'")
 
         if not gagent_start_cmd:
             return
 
-        error.context("Try to start 'qemu-guest-agent'.", logging.info)
+        logging.info("Try to start 'qemu-guest-agent'.")
         session = self._get_session(params, vm)
         s, o = self._session_cmd_close(session, gagent_start_cmd)
         if s and "already been started" not in o:
-            raise error.TestFail("Could not start qemu-guest-agent in VM"
-                                 " '%s', detail: '%s'" % (vm.name, o))
+            test.fail("Could not start qemu-guest-agent in VM"
+                      " '%s', detail: '%s'" % (vm.name, o))
 
-    @error.context_aware
-    def gagent_create(self, params, vm, *args):
+    def gagent_create(self, params, test, vm, *args):
```

Comment:
Why pass `test` here? You have it in your class, just use self.test.error() directly is okay.

Reply:
Added "self.test = test" to __init__ function in class QemuGuestAgentTest,  no longer passing test parameter in function. e.g  gagent_install  gagent_start....   and so on. 

 (1/1) smp_2.512m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.7.4.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_reset_reboot_shutdown_fsfreeze: PASS (446.54 s)

Reply:
@vivianQizhu  I use "super(QemuGuestAgentTest, self).__init__(self, test, params, env)" to replace self.test = test. 

 (1/1) smp_2.512m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.RHEL.7.4.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_reset_reboot_shutdown_fsfreeze: PASS (442.66 s)